File: 0242-valid-anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        f_word, s_word = {}, {}

        for i in range(len(s)):
            f_word[s[i]] = 1 + f_word.get(s[i], 0)
            s_word[t[i]] = 1 + s_word.get(t[i], 0)
        
        return f_word == s_word

File: 0242-valid-anagram/solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> m = new HashMap<>();

        for (char x : s.toCharArray()) {
            m.put(x, m.getOrDefault(x, 0) + 1);
        }

        for (char x : t.toCharArray()) {
            m.put(x, m.getOrDefault(x, 0) - 1);
        }

        for (int val : m.values()) {
            if (val != 0) {
                return false;
            }
        }

        return true;
    }
}
